File: Iverson.Clients/Java/conformance/src/main/java/io/iverson/conformance/models/SharedAuthor.java
package io.iverson.conformance.models;

import io.iverson.client.annotations.IversonEntity;
import io.iverson.client.annotations.IversonKey;
import io.iverson.client.annotations.IversonTenant;

import java.util.UUID;

/**
 * S4 {@code interop}'s "one" side. Every one of the five drivers declares the same type name and
 * shape; only the .NET driver ever registers it (register-once rule), so this driver's own
 * {@code SchemaRegistrar} is never invoked for it.
 */
@IversonEntity
public class SharedAuthor {

    @IversonKey
    private UUID id;

    @IversonTenant
    private String tenantId;

    private String ownerId;
    private String name;

    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }

    public String getTenantId() { return tenantId; }
    public void setTenantId(String tenantId) { this.tenantId = tenantId; }

    public String getOwnerId() { return ownerId; }
    public void setOwnerId(String ownerId) { this.ownerId = ownerId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
}

File: Iverson.Clients/DotNet/Iverson.Client.Conformance.Driver/Models/SharedAuthor.cs
using Iverson.Client.Attributes;

namespace Iverson.Client.Conformance.Driver.Models;

/// <summary>
/// S4 <c>interop</c>'s "one" side. Registered only once, by the .NET driver — see
/// <c>Scenarios/InteropScenario.cs</c> for why five registrations of the same type name would
/// silently overwrite one another.
/// </summary>
[IversonEntity]
public class SharedAuthor
{
    [IversonKey] public Guid Id { get; set; }
    public string TenantId { get; set; } = string.Empty;
    public string OwnerId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
}
